src/en/darthsdroids/src/eu/kanade/tachiyomi/extension/en/darthsdroids/DarthsDroids.kt

```diff
@@ -0,0 +1,222 @@
+// If possible, i would use the Zip downloads to source finished
+// chapters. However, as far as i can see, there is no way to hack
+// this into the existing extension API.
+class DarthsDroids : HttpSource() {
+    override val name = "Darths & Droids"
+    override val baseUrl = "https://www.darthsanddroids.net/"
+    override val lang = "en"
+    override val supportsLatest = false
+
+    private val mainArchiveUrl = "$baseUrl/archive.html"
+
+    private fun dndThumbnailUrlForTitle(nthManga: Int): String = when (nthManga) {
+        0 -> "https://www.darthsanddroids.net/cast/QuiGon.jpg" // D&D1
+        1 -> "https://www.darthsanddroids.net/cast/Anakin2.jpg" // D&D2
+        2 -> "https://www.darthsanddroids.net/cast/ObiWan3.jpg" // D&D3
+        3 -> "https://www.darthsanddroids.net/cast/JarJar2.jpg" // JJ
+        4 -> "https://www.darthsanddroids.net/cast/Leia4.jpg" // D&D4
+        5 -> "https://www.darthsanddroids.net/cast/Han5.jpg" // D&D5
+        6 -> "https://www.darthsanddroids.net/cast/Luke6.jpg" // D&D6
+        7 -> "https://www.darthsanddroids.net/cast/Cassian.jpg" // R1
+        8 -> "https://www.darthsanddroids.net/cast/C3PO4.jpg" // Muppets
+        9 -> "https://www.darthsanddroids.net/cast/Finn7.jpg" // D&D7
+        10 -> "https://www.darthsanddroids.net/cast/Han4.jpg" // Solo
+        11 -> "https://www.darthsanddroids.net/cast/Hux8.jpg" // D&D8
+        // Just some nonsense fallback that screams »Star Wars« but is also so recognisably
+        // OT that one can understand it’s a mere fallback. Better thumbnails require an
+        // extension update.
+        else -> "https://www.darthsanddroids.net/cast/Vader4.jpg"
+    }
+
+    private fun dndManga(archiveUrl: String, mangaTitle: String, mangaStatus: Int, nthManga: Int): SManga = SManga.create().apply {
+        url = archiveUrl
+        thumbnail_url = dndThumbnailUrlForTitle(nthManga)
+        title = mangaTitle
+        author = "David Morgan-Mar & Co."
+        artist = "David Morgan-Mar & Co."
+        description = """What if Star Wars as we know it didn't exist, but instead the
+            | plot of the movies was being made up on the spot by players of
+            | a Tabletop Game?\n
+            | Well, for one, the results might actually make a lot more sense,
+            | from an out-of-story point of view…
+        """.trimMargin()
+        genre = "Campaign Comic, Comedy, Space Opera, Science Fiction"
+        status = mangaStatus
+        update_strategy = when (mangaStatus) {
+            SManga.COMPLETED -> UpdateStrategy.ONLY_FETCH_ONCE
+            else -> UpdateStrategy.ALWAYS_UPDATE
+        }
+        initialized = true
+    }
+
+    override fun fetchPopularManga(page: Int): Observable<MangasPage> {
+        val mainArchive = client.newCall(GET(mainArchiveUrl, headers)).execute().asJsoup()
+        val archiveData = mainArchive.select("div.text > table.text > tbody > tr")
+
+        val mangas = mutableListOf<SManga>()
```

Comment:
Avoid mutable structures if possible (probably easier if you migrate away from the `fetch*` methods).